### PR TITLE
Use the Az module instead of AzureRM

### PR DESCRIPTION
**Pull request by paddy74**
AzureRM is deprecated and not compatible with PowerShell 7. As this script is referenced in official documentation, it should be updated.
This commit also updates variable names where they are retrieved such as `.SubscriptionName` to `.DisplayName` where the former no longer exists.

### CODE REVIEW

File: spncreation.ps1 [hunk 1]
```diff
@@ -41,57 +41,57 @@ function Get-Password
...
 }
 
-Import-Module -Name AzureRM.Profile
-Write-Output "Provide your credentials to access Azure subscription $subscriptionName" -Verbose
-Login-AzureRmAccount -SubscriptionName $subscriptionName -EnvironmentName $environmentName
-$azureSubscription = Get-AzureRmSubscription -SubscriptionName $subscriptionName
-$connectionName = $azureSubscription.SubscriptionName
+Import-Module -Name Az.Accounts
+Write-Output "Provide your credentials to access your Azure subscription $subscriptionName" -Verbose
+Login-AzAccount -SubscriptionName $subscriptionName -EnvironmentName $environmentName
```
Comment (dcaro) [inline spncreation.ps1]:
```suggestion
Connect-AzAccount -SubscriptionName $subscriptionName -EnvironmentName $environmentName
```

File: spncreation.ps1 [hunk 2]
```diff
@@ -41,57 +41,57 @@ function Get-Password
...
 $tenantId = $azureSubscription.TenantId
 $id = $azureSubscription.SubscriptionId
 
 
-#Create a new AD Application
+# Create a new AD Application
 Write-Output "Creating a new Application in AAD (App URI - $identifierUri)" -Verbose
 $password = Get-Password
-$azureAdApplication = New-AzureRmADApplication -DisplayName $displayName -HomePage $homePage -IdentifierUris $identifierUri -Password $password -Verbose
+$azureAdApplication = New-AzADApplication -DisplayName $displayName -HomePage $homePage -IdentifierUris $identifierUri -Password $password -Verbose
```
Comment (dcaro) [inline spncreation.ps1]: This command will not work with Az PowerShell 7 and above. It is using Microsoft Graph sot the password cannot be set and the identifier Uri must use a verified domain in of the organization or its subdomain.

```suggestion
$azureAdApplication = New-AzADApplication -DisplayName $displayName -HomePage $homePage -Verbose
```

File: spncreation.ps1 [hunk 3]
```diff
@@ -41,57 +41,57 @@ function Get-Password
...
 $appId = $azureAdApplication.ApplicationId
```
Comment (dcaro) [inline spncreation.ps1]:
```suggestion
$appId = $azureAdApplication.AppId
```

File: spncreation.ps1 [hunk 4]
```diff
@@ -41,57 +41,57 @@ function Get-Password
...
 Write-Output "SPN creation completed successfully (SPN Name: $spnName)" -Verbose
 
 
-#Assign role to SPN
+# Assign role to SPN
 Write-Output "Waiting for SPN creation to reflect in Directory before Role assignment"
 Start-Sleep 20
 Write-Output "Assigning role ($spnRole) to SPN App ($appId)" -Verbose
-New-AzureRmRoleAssignment -RoleDefinitionName $spnRole -ServicePrincipalName $appId
+New-AzRoleAssignment -RoleDefinitionName $spnRole -ServicePrincipalName $appId
```
Comment (dcaro) [inline spncreation.ps1]:
```suggestion
New-AzRoleAssignment -RoleDefinitionName $spnRole -ServicePrincipalName $spn.AppId
```

File: spncreation.ps1 [hunk 5]
```diff
@@ -15,20 +15,20 @@ param
...
     }
-    return (Get-Module Azure -ListAvailable).Version
+    return (Get-Module Az -ListAvailable).Version
 }
 
 function Get-Password
 {
     $currentAzurePSVersion = Get-AzureCmdletsVersion
-    $azureVersion511 = New-Object System.Version(5, 1, 1)
+    $minAzurePSVersion = New-Object System.Version(5, 6, 0)
```
Comment (dcaro) [inline spncreation.ps1]: Az PowerShell version 8 is now available, I would recommend considering the most recent version of this module for this script.
```suggestion
    $minAzurePSVersion = New-Object System.Version(8, 0, 0)
```

**Review by dcaro — COMMENTED** [review]
I made some comments to make this PR compatible with Az PowerShell version 8.

**Comment by stevevi** [comment]
Thank you @paddy74 and @dcaro for your input.  I updated the article and moved the PS script into the main body of the article for easier maintenance.

https://docs.microsoft.com/en-us/azure/azure-government/connect-with-azure-pipelines#generate-a-service-principal

This standalone dev repo is no longer referenced from the updated article.  Feel free to close the PR.  Also, my apologies @paddy74 that it took such a long time to act on your initial PR.  The software dev who originally wrote the script changed roles and this repo was no longer monitored actively.

**Comment by paddy74** [comment]
@dcaro I've applied the recommended changes.
Although this repository is no longer monitored or referenced by the documentation, individuals may still end up referencing it. As such, I recommend these changes be merged.
A separate PR should be opened to indicate in the `README.md` that this repository is deprecated and to direct users to the documentation. If possible, I also recommend the "deprecated" tag be added to this repository.

**Comment by stevevi** [comment]
Thanks @paddy74 for your input.  I'll defer to @yujhongmicrosoft who is the repo owner for further action.

**Pull request closed without merging**